### PR TITLE
Add implementation of TCN and TempCNN

### PR DESCRIPTION
**Pull request by devisperessutti**
this MR adds implementation of:

 * [TCN](https://arxiv.org/pdf/1803.01271.pdf) taken from [keras-TCN](https://github.com/philipperemy/keras-tcn)
 * [TempCNN](https://arxiv.org/pdf/1811.10166.pdf) taken from [temporalCNN](https://github.com/charlotte-pel/temporalCNN)

### CODE REVIEW

File: eoflow/models/layers.py [hunk 1]
```diff
@@ -1,7 +1,139 @@
...
+            training: boolean indicating whether the layer should behave in training mode or in inference mode
+            dilation_rate: The dilation power of 2 we are using for this residual block
+            nb_filters: The number of convolutional filters to use in this block
+            kernel_size: The size of the convolutional kernel
+            padding: The padding used in the convolutional layers, 'same' or 'causal'.
+            activation: The final activation used in o = Activation(x + F(x))
+            dropout_rate: Float between 0 and 1. Fraction of the input units to drop.
+            kernel_initializer: Initializer for the kernel weights matrix (Conv1D).
+            use_batch_norm: Whether to use batch normalization in the residual layers or not.
+            use_layer_norm: Whether to use layer normalization in the residual layers or not.
```
Comment (veseln) [inline eoflow/models/layers.py]: add docstring for last_block parameter

File: eoflow/models/classification_temp_nets.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                                description='Padding type used in convolutions.')
+        use_skip_connections = fields.Bool(missing=True, description='Flag to whether to use skip connections.')
+        return_sequences = fields.Bool(missing=False, description='Flag to whether return sequences or not.')
+        activation = fields.Str(missing='linear', description='Activation function used in final filters.')
+        kernel_initializer = fields.Str(missing='he_normal', description='method to initialise kernel parameters.')
+
+        use_batch_norm = fields.Bool(missing=False, description='Whether to use batch normalisation.')
+        use_layer_norm = fields.Bool(missing=False, description='Whether to use layer normalisation.')
+
+        class_weights = fields.List(fields.Float, missing=None, description='Class weights used in training.')
```
Comment (veseln) [inline eoflow/models/classification_temp_nets.py]: This should be deleted cause it's defined in the base classfication schema.

File: eoflow/models/classification_temp_nets.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        nb_fc_stacks = fields.Int(missing=1, description='Number of fully connected layers.')
+        padding = fields.String(missing='SAME', validate=OneOf(['SAME']),
+                                description='Padding type used in convolutions.')
+        activation = fields.Str(missing='relu', description='Activation function used in final filters.')
+        kernel_initializer = fields.Str(missing='he_normal', description='Method to initialise kernel parameters.')
+        kernel_regularizer = fields.Float(missing=1e-6, description='L2 regularization parameter.')
+
+        use_batch_norm = fields.Bool(missing=False, description='Whether to use batch normalisation.')
+
+        class_weights = fields.List(fields.Float, missing=None, description='Class weights used in training.')
```
Comment (veseln) [inline eoflow/models/classification_temp_nets.py]: Same as above.

File: eoflow/models/layers.py (hunk 1 above)
Comment (devisperessutti) [reply eoflow/models/layers.py]: added

File: eoflow/models/classification_temp_nets.py (hunk 2 above)
Comment (devisperessutti) [reply eoflow/models/classification_temp_nets.py]: removed

File: eoflow/models/classification_temp_nets.py (hunk 3 above)
Comment (devisperessutti) [reply eoflow/models/classification_temp_nets.py]: removed

**Pull request merged**